job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/QueueServices.java

```diff
@@ -67,44 +63,14 @@ public QueueServices(final Connection connection, final Channel publisherChannel
      */
     public void sendMessage(
         final String partitionId, final String jobId, final WorkerAction workerAction
-    ) throws IOException, URISyntaxException {
+    ) throws IOException {
         //  Generate a random task id.
         LOG.debug("Generating task id ...");
         final String taskId = UUID.randomUUID().toString();
 
-        //  Serialise the data payload. Encoding type is provided in the WorkerAction.
-        final byte[] taskData;
-
         //  Check whether taskData is in the form of a string or object, and serialise/decode as appropriate.
         LOG.debug("Validating the task data ...");
         final Object taskDataObj = workerAction.getTaskData();
-        
-        if (taskDataObj instanceof String) {
```

Comment:
You're using `taskDataObj` directly now but this code suggests that it could be a string, so that should be handled.  We should probably still go through a conversion, although of course it will be a different conversion than the one that used to be in this branch that was converting to a `byte[]`.

Reply:
Done, note that I extracted the method for better readability, let me know if you want me to restore.